### PR TITLE
Add comprehensive tests to test the kernel across available dtypes.

### PR DESCRIPTION
**Pull request by pashu123**
Added softmax and gemm kernel to test across the available float and int dtypes.

### CODE REVIEW

File: core/tests/kernel/coverage_test.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import torch
+import shark_turbine.kernel as tk
+import shark_turbine.kernel.lang as tkl
+import pytest
+
+
+TKL_TO_TORCH_DTYPE = {
```
Comment (Groverkss) [inline core/tests/kernel/coverage_test.py:7]: Can we add this to DataType as a method? a from_torch_type and to_torch_type would be good to have.

Comment (pashu123) [reply core/tests/kernel/coverage_test.py]: Done. PTAL.

**Comment by dan-garvey** [comment]
is this stale?

**Comment by pashu123** [comment]
> is this stale?

@Groverkss Is this required?
